src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdIT.java

```diff
@@ -0,0 +1,82 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import pl.byczazagroda.trackexpensesappbackend.BaseIntegrationTestIT;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransaction;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+public class GetFinancialTransactionByWalletIdIT extends BaseIntegrationTestIT {
+
+    @Autowired
+    FinancialTransactionRepository financialTransactionRepository;
+
+    @Autowired
+    WalletRepository walletRepository;
+
+    @BeforeEach
+    public void clearTestDB() {
+        financialTransactionRepository.deleteAll();
+        walletRepository.deleteAll();
+    }
+
+    @Test
+    @DisplayName("when wallet id is correct returns List of financial transactions DTO related to wallet")
+    public void givenValidWalletId_whenGetFinancialTransactionsByWalletId_thenCorrectResponse() throws Exception {
+        Wallet wallet = createTestWallet();
+        FinancialTransaction financialTransaction = createTestFinancialTransaction(wallet);
+        mockMvc.perform(get("/api/transactions")
+                .contentType(MediaType.APPLICATION_JSON)
+                .queryParam("walletId", String.valueOf(wallet.getId())))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(jsonPath("$.[0].id").value(financialTransaction.getId()))
+                .andExpect(jsonPath("$.[0].amount").value(financialTransaction.getAmount()))
+                .andExpect(jsonPath("$.[0].description").value(financialTransaction.getDescription()))
+                .andExpect(jsonPath("$.[0].type").value(financialTransaction.getType().name()))
+                .andExpect(jsonPath("$.[0].date").value(financialTransaction.getDate().toString()));
+
+        Assertions.assertEquals(1, financialTransactionRepository.count());
+        Assertions.assertEquals(1, walletRepository.count());
+    }
+
+    @Test
+    @DisplayName("when wallet id is correct returns error response dto and has 404 status code")
+    public void givenInvalidWalletId_whenGetFinancialTransactionsByWalletId_thenNotFoundStatusCode() throws Exception {
+        mockMvc.perform(get("/api/transactions")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .queryParam("walletId", "1"))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+
```

Comment:
From a business point of view, the status returned in the response body is important. Add a check that this status has the correct value W003.
```json
{
    "status": "W003",
    "message": "WALLET_NOT_FOUND",
    "description": "Wallet with id: 38 does not exist",
    "statusCode": 404
}
```